lib/active_job/workers_relation.rb

```diff
@@ -0,0 +1,61 @@
+# A relation of workers.
+#
+# Relations are enumerable, so you can use +Enumerable+ methods on them.
+# Notice however that using these methods will imply loading all the relation
+# in memory, which could introduce performance concerns.
+#
+# Internally, +ActiveJob+ will always use paginated queries to the underlying
+# queue adapter. The page size can be controlled via +config.active_job.default_page_size+
+# (1000 by default).
+class ActiveJob::WorkersRelation
+  include Enumerable
+
+  attr_reader :default_page_size
+  attr_accessor :offset_value, :limit_value
+
+  delegate :last, :[], :count, :empty?, :length, :size, :to_s, :inspect, :reverse, to: :to_a
+
+  ALL_WORKERS_LIMIT = 100_000_000 # When no limit value it defaults to "all workers"
+
+  def initialize(workers: [], default_page_size: ActiveJob::Base.default_page_size)
```

Comment:
The idea here won't be to get the workers already loaded and do all the limiting and offsetting in memory because that'd prevent us from taking advantage of the database for this when the adapter supports that. Just like with the `JobsRelation`, we'd want to get an adapter here, to which we can delegate the fetching of the workers, with the configured `limit` and `offset`. 

I also don't think we need to worry about having so many workers that we have to pass a page size to do an internal pagination. This is necessary with jobs because we might want to load thousands of jobs to filter, but we don't want to request all of them at once from whatever underlying store we're using. This means you can spare all of the `config.active_job.default_page_size` comments on top and as a parameter. In fact, it's not used anywhere here. 

In this way, you'd have something like this: 
```ruby
  def initialize(queue_adapter:)
    @queue_adapter = queue_adapter

    set_defaults
  end

  def offset(offset)
    clone_with offset_value: offset
  end

  def limit(limit)
    clone_with limit_value: limit
  end

  def each(&block)
    workers.each(&block)
  end

  private
     def workers
       @workers ||= queue_adapter.fetch_workers(self)
     end
```

And if the adapter exposes workers, it'll support a `fetch_workers` method that will use the `limit` and `offset` of the relation to fetch these workers. Each adapter will do it in its own way, and we'd also want to wrap the workers in `MissionControl::Jobs::Worker` objects from here, to behave in the same way as the jobs related methods, that wrap the results in `ActiveJob::JobProxy` objects. Solid Queue would do something like: 
```ruby
  def fetch_workers(workers_relation)
    SolidQueue::Process.where(kind: "Worker").offset(workers_relation.offset).limit(workers_relation.limit_value).collect do |process| 
      worker_from_solid_queue_process(process)
    end
  end

# ...

  private
    def worker_from_solid_queue_process(process)
       MissionControl::Jobs::Worker.new(queue_adapter: self, **worker_attributes_from_solid_queue_process(process))
    end
```
replacing [the current `#workers` method](https://github.com/basecamp/mission_control-jobs/blob/9e07a5d262cfc837624ff84f9944a41fa61c1bb0/lib/active_job/queue_adapters/solid_queue_ext.rb#L49-L53) in the extension to the Solid Queue adapter. 

Finally, to glue everything together, instead of having the `server` provide the workers directly, it'd need to provide a `workers_relation` that we'd pass to the `Page`. 

```ruby
module MissionControl::Jobs::Server::Workers
  def workers_relation
    MissionControl::Jobs::WorkersRelation.new(queue_adapter: queue_adapter)
  end
```

And then we'd use that in the controllers. After this, there are some improvements that could be done but that should be enough for pagination and this PR 😊